app/models/file_upload/Nonce.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.file_upload
+
+import play.api.libs.json.Format
+
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+import scala.language.implicitConversions
+import scala.util.{Random, Try}
+
+/** Random integer value container. */
+sealed trait Nonce {
+  val value: Int
+}
+
+object Nonce {
+
+  final def random: Nonce =
+    toNonce(Random.nextInt())
+
+  final def apply(value: Int): Nonce =
+    toNonce(value)
+
+  object Any extends Nonce {
```

Comment:
For Any to be a companion object needs to be declared in same file as class Any.